server/app/mutations/stacks/restart.rb

```diff
@@ -0,0 +1,17 @@
+module Stacks
+  class Restart < Mutations::Command
+
+    required do
+      model :stack, class: Stack
+    end
+
+    def execute
+      self.stack.grid_services.each do |service|
+        outcome = GridServices::Restart.run(grid_service: service)
+        unless outcome.success?
+          add_error(service.to_path, :stop_failed, outcome.errors.message)
```

Comment:
Use `Stacks::Common` and `handle_service_outcome_errors`?

Impossible to test though, because the `GridServices::Restart` mutation is async and doesn't return any errors... in fact, it doesn't even log them (#2126).